File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.js
function solution(sides) {
    var answer = 0;
    var max = 0; // 가장 긴 변
    var num = 0; // 가장 긴 변의 될 수 있는 나머지 한 변의 개수
    var count = 0; // 나머지 한 변이 가장 긴 변인 경우 될 수 있는 한 변
    
    // sides의 길이는 2
    // sides의 요소 중 큰 요소 = max, 나머지 요소 = num
    for (let i = 0; i < sides.length; i++) {
        if (sides[0] < sides[1]) {
            max = sides[1];
            num = sides[0];
        } else {
            max = sides[0];
            num = sides[1];
        }
        
        // 나머지 한 변이 2일 경우 1 리턴
        answer = num <= 1 ? 1 : sides[0] + sides[1];
    }
    
    // 가장 긴 변의 길이는 다른 두 변의 길이의 합보다 작아야 함
    if (answer != 1) {
        for (let i = max + 1; i < (max + num); i++) {
            count++;
            answer = num + count;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.java
import java.util.*;

class Solution {
    public int solution(int[] sides) {
        // 가장 긴 변이 sides[1]인 경우 될 수 있는 나머지 한 변의 개수 = sides[0];
        // 나머지 한 변이 가장 긴 변인 경우 = (sides[1] + 1) < (sides[0] + sides[1])
        // return = (sides[1] + 1) < (sides[0] + sides[1]) 만족하는 수의 개수 + sides[0]
        
        Arrays.sort(sides); // 오름차순 정렬, sides의 길이는 2
        
        int max = sides[0] + sides[1]; // 가장 긴 변의 길이 < 다른 두 변의 길이의 합
        int num = max - (sides[1] + 1); // 나머지 한 변이 가장 긴 변인 경우될 수 있는 한 변의 개수
        int answer = 0;
        
        if (sides[1] == 2) {
            // 두 변이 1, 2 인 경우
            answer = 1;
        } else {
            answer = sides[0] + num;
        }
        
        return answer;
    }
}